batch/batch/worker_config.py

```diff
@@ -0,0 +1,131 @@
+#   preemptible: bool
+# boot_disk: dict
+#   project: str
+#   zone: str
+#   type: str (pd-ssd, pd-standard)
+#   size: int (in GB)
+#   image: str
+
+
+class WorkerConfig:
+    @staticmethod
+    def from_instance_config(instance_config):
+        instance_project, instance_zone, instance_family, instance_type, instance_cores = \
+            parse_machine_type_str(instance_config['machineType'])
+
+        preemptible = instance_config['scheduling']['preemptible']
+
+        boot_disk = [disk_config for disk_config in instance_config['disks']
+                     if disk_config['boot']][0]
+
+        boot_disk_project, boot_disk_zone, boot_disk_type = parse_disk_type(boot_disk['initializeParams']['diskType'])
+        boot_disk_image = boot_disk['initializeParams']['sourceImage']
+        boot_disk_size = int(boot_disk['initializeParams']['diskSizeGb'])
+
+        config = {
+            'version': WORKER_CONFIG_VERSION,
+            'instance': {
+                'project': instance_project,
+                'zone': instance_zone,
+                'family': instance_family,
+                'type': instance_type,
+                'cores': int(instance_cores),
+                'preemptible': preemptible
+            },
+            'boot_disk': {
+                'project': boot_disk_project,
+                'zone': boot_disk_zone,
+                'type': boot_disk_type,
+                'size': boot_disk_size,
+                'image': boot_disk_image
+            }
+        }
+
+        return WorkerConfig(config)
+
+    def __init__(self, config):
+        self.config = config
+
+        self.version = self.config['version']
+        assert self.version == 1
+
+        instance = self.config['instance']
+        boot_disk = self.config['boot_disk']
+
+        self.instance_project = instance['project']
+        self.instance_zone = instance['zone']
+        self.instance_family = instance['family']
+        self.instance_type = instance['type']
+
+        self.cores = instance['cores']
+        self.preemptible = instance['preemptible']
+        self.boot_disk_size_gb = boot_disk['size']
+        self.boot_disk_type = boot_disk['type']
+
+    def is_valid_configuration(self):
+        is_valid = True
+        dummy_resources = self.resources(0, 0)
+        for resource in dummy_resources:
+            is_valid &= resource['name'] in RATES
+        return is_valid
+
+    def resources(self, cpu_in_mcpu, memory_in_bytes):
+        resources = []
+
+        preemptible = 'preemptible' if self.preemptible else 'nonpreemptible'
+        worker_fraction = cpu_in_mcpu / (self.cores * 1000)
+
+        resources.append({'name': f'compute/{self.instance_family}-{self.instance_type}-{preemptible}/1',
+                          'quantity': cpu_in_mcpu})
+
+        resources.append({'name': f'memory/{self.instance_family}-{self.instance_type}-{preemptible}/1',
+                          'quantity': math.ceil(memory_in_bytes / 1024 / 1024)})
```

Comment:
these should be 1000.

Reply:
All of our machines' memory in bytes should be cleanly divided by 1_000_000, right? We should assert that and use `memory_in_bytes // 1_000_000`.

Reply:
no ceil.